reader-osm/src/test/java/com/graphhopper/GraphHopperTest.java

```diff
@@ -1742,11 +1781,13 @@ public void testTurnCostsOnOff() {
         final String profile2 = "profile_turn_costs";
         final String vehicle = "car";
         final String weighting = "fastest";
-        GraphHopper hopper = createGraphHopper("car|turn_costs=true").
+        GraphHopper hopper = new GraphHopper().
+                setGraphHopperLocation(GH_LOCATION).
                 setOSMFile(MOSCOW).
+                // add profile with turn costs first when no flag encoder is explicitly added
```

Comment:
It can be done without this requirement but the check in EncodingManager will be more complicated. E.g. loop through all profiles and add the vehicles for profiles with turn costs first, then without turn costs later.

Reply:
Hm, but this seems dangerous then. It is too easy to add the profiles in the wrong order? I think either we should handle this in GraphHopper, or at least throw an error in case this constraint is violated.

Reply:
Ok, will try to fix this in GraphHopper